props/saber_fett263_buttons.h

```diff
@@ -355,6 +1485,308 @@ SaberFett263Buttons() : PropBase() {}
+    EDIT_WHITE,
+    ZOOM_COLOR,
+#endif
+  };
+
+#define CC_NEW_COLOR(N) NewColor(N,1);
+
+// Update Style
+  virtual void UpdateStyle(int preset_num) {
+    TRACE(PROP, "start");
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+#define UNSET_BLADE_STYLE(N) \
+    delete current_config->blade##N->UnSetStyle();
+    ONCEPERBLADE(UNSET_BLADE_STYLE)
+    current_preset_.SetPreset(preset_num);
+#ifdef DYNAMIC_BLADE_LENGTH
+    savestate_.ReadINIFromSaveDir("curstate");
+#define WRAP_BLADE_SHORTERNER(N) \
+    if (savestate_.blade##N##len != -1 && savestate_.blade##N##len != current_config->blade##N->num_leds()) { \
+      tmp = new BladeShortenerWrapper(savestate_.blade##N##len, tmp);	\
+    }
+#else
+#define WRAP_BLADE_SHORTERNER(N)
+#endif
+
+    
+#define SET_BLADE_STYLE(N) do {						\
+    BladeStyle* tmp = style_parser.Parse(current_preset_.current_style##N.get()); \
+    WRAP_BLADE_SHORTERNER(N)                                            \
+    current_config->blade##N->SetStyle(tmp);				\
+  } while (0);
+
+    ONCEPERBLADE(SET_BLADE_STYLE)
+
+#ifdef SAVE_COLOR_CHANGE
+    SaberBase::SetVariation(current_preset_.variation);
+#else
+    SaberBase::SetVariation(0);
+#endif
+    TRACE(PROP, "end");
+  }
+
+// Update Font/ Save Style
+  virtual void UpdateFont(int preset_num) {
+    TRACE(PROP, "start");
+    bool on = SaberBase::IsOn();
+    if (on) Off();
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+#define UNSET_BLADE_STYLE(N) \
+    delete current_config->blade##N->UnSetStyle();
+    ONCEPERBLADE(UNSET_BLADE_STYLE)
+    current_preset_.SetPreset(preset_num);
+#define SET_BLADE_STYLE(N) \
+    current_config->blade##N->SetStyle(style_parser.Parse(current_preset_.current_style##N.get()));
+    ONCEPERBLADE(SET_BLADE_STYLE)
+    chdir(current_preset_.font.get());
+#ifdef SAVE_COLOR_CHANGE
+    SaberBase::SetVariation(current_preset_.variation);
+#else
+    SaberBase::SetVariation(0);
+#endif
+    if (on) {
+#ifdef FETT263_EDIT_MODE_MENU
+      if (menu_type_ == MENU_PREON_OPTION || menu_type_ == MENU_PREON_SIZE) {
+        On();
+      } else {
+        FastOn();
+      }
+#else
+      FastOn();
+#endif      
+    }
+    TRACE(PROP, "end");
+  }
+
+  bool Parse(const char *cmd, const char* arg) override {
+    if (PropBase::Parse(cmd, arg)) return true;
+    if (!strcmp(cmd, "list_current_tracks")) {
+      // Tracks must be in: font/tracks/*.wav 
+      LOCK_SD(true);
+      for (const char* dir = current_directory; dir; dir = next_current_directory(dir)) {
+        PathHelper path(dir, "tracks");
+        ListTracks(path);
+      }
+      LOCK_SD(false);
+      return true;
+    }
+    return false;
+  }
+
+  // Go to first Preset.
+  virtual void first_preset() {
+#ifdef SAVE_PRESET
+    SaveState(0);
+#endif
+    UpdateFont(0);
+}
+
+  // Go to the next Preset with update.
```

Comment:
I don't understand what "with update" means.

Reply:
was UpdateFont, the usage is to bypass Preon Effects with FastOn within the UpdateFont function, comment updated